tests/test_gates_gates.py

```diff
@@ -414,6 +421,15 @@ def test_u3(backend):
 
     backend.assert_allclose(final_state, target_state)
 
+    # testing random expectation value due to global phase difference
+    observable = random_hermitian(2**nqubits, backend=backend)
```

Comment:
Better adding a seed, otherwise the test would lack reproducibility.

Reply:
The point of not adding a seed is that this test should be observable-independent. It has to work for all observables, a fixed one might introduce bias.

Reply:
> The point of not adding a seed is that this test should be observable-independent. It has to work for all observables, a fixed one might introduce bias. One option would be testing with multiple seeds, but that still doesn't eliminate the bias.

I understood the point, that's the usual one that motivates random numbers in the unit tests.
But this partially defeats the point of unit tests, since in this way it will become not reproducible.

If you want to test multiple observables, you should pick a single seed, and just extract multiple times. In this way, you will get N random observables, but they have to be always the same for all the runs.

The rationale is that if at some points something goes wrong, you want to jump back and forth and check where it stopped working. But if it's only failing sometimes for some of them, it becomes quite difficult to debug.

Reply:
Seed(s) introduced.